### PR TITLE
feat(kafka): add Inventory update events consumer

### PR DESCRIPTION
**Pull request by petrblaho**
**feat(kafka): add Inventory update events consumer [[RHINENG-23280](https://redhat.atlassian.net/browse/RHINENG-23280)]**

## Description
   Implements the Kafka consumer logic for Inventory `created`
 and `updated` events to populate the `systems` table,
 replacing the legacy Cyndi replication.

### Key Changes
   - **Model:** Added `KafkaSystem` to safely write to the
 `systems` table (bypassing the read-only `V2::System` view
 during migration).
   - **Service:** Created `Kafka::SystemImporter` to map event
 fields and perform safe `upsert` operations.
   - **Routing:** `InventoryEventsConsumer` now routes events
 to `SystemImporter` (and `PolicySystemImporter` when a policy
 ID is present).
   - **Protection:** Added staleness checks (drops
 out-of-order messages based on `updated` timestamp) and
 graceful error handling to prevent Karafka loop crashes.
   - **Testing:** Added a dedicated `kafka_system` factory and
 comprehensive RSpec coverage for all edge cases.

## Jira Issue
   - [RHINENG-23280](https://issues.redhat.com/browse/RHINENG-23280)

## AI/LLM Usage
   This PR was authored with the assistance of an AI coding
 agent (Gemini 3.1 Pro Preview via pi/unipi) to scaffold the
 `Kafka::SystemImporter` upsert logic, the `KafkaSystem`
 temporary model, inventory event routing, and comprehensive
 RSpec test suites including factory setups. All AI output was
 reviewed, tested, and modified by the human author.

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [x] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

[RHINENG-23280]: https://redhat.atlassian.net/browse/RHINENG-23280?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RHINENG-23280]: https://redhat.atlassian.net/browse/RHINENG-23280?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- Add temporary `KafkaSystem` ActiveRecord model (backed by `systems`) to ingest inventory data directly during migration from legacy Cyndi replication
- Introduce `Kafka::SystemImporter` to validate Kafka payloads and safely upsert `KafkaSystem`/`systems` with:
  - out-of-order protection via `updated` timestamp staleness checks
  - logging + swallow-on-failure error handling (no Karafka crash)
- Update `InventoryEventsConsumer` routing so `inventory` `created`/`updated` events always go through `Kafka::SystemImporter`, with `Kafka::PolicySystemImporter` only invoked when `policy_id` is present
- Add `:kafka_system` FactoryBot factory to isolate importer tests from legacy associations/side effects
- Expand RSpec coverage for importer validation, staleness rejection, optional-field defaults (e.g., `groups`), routing edge cases, and upsert/DB constraint error handling

### CODE REVIEW

File: app/services/kafka/system_importer.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    end
+
+    def import
+      payload = @message['host'] || @message
+      unless valid_payload?(payload)
+        @logger.error('[Kafka::SystemImporter] Ignored invalid message: missing host id')
+        return
+      end
+
+      id = payload['id']
```
Comment (romanblanco) [inline app/services/kafka/system_importer.rb]: nitpick: `id, updated = payload.values_at('id', 'updated')`

**Review by romanblanco — APPROVED** [review]

**Comment by petrblaho** [comment]
superseded by https://github.com/RedHatInsights/compliance-backend/pull/2826
